tests/e2e/main_test.go

```diff
@@ -77,3 +84,110 @@ func TestLintMetrics(t *testing.T) {
 		t.Fatalf("the problems encountered in Lint are: %v", problems)
 	}
 }
+
+func TestDocumentation(t *testing.T) {
+	labelsDocumentation, err := getLabelsDocumentation()
+	if err != nil {
+		t.Fatal("Cannot get labels documentation", err)
+	}
+
+	buf := &bytes.Buffer{}
+
+	err = framework.KsmClient.metrics(buf)
+	if err != nil {
+		t.Fatal("Failed to get metrics from kube-state-metrics", err)
+	}
+
+	parser := &expfmt.TextParser{}
+	metricFamilies, err := parser.TextToMetricFamilies(buf)
+	if err != nil {
+		t.Fatal("Cannot decode metrics", err)
+	}
+
+	for _, metricFamily := range metricFamilies {
+		metric := metricFamily.GetName()
+
+		acceptedLabelNames, ok := labelsDocumentation[metric]
+		if !ok {
+			t.Errorf("Metric %v not found in documentation.", metric)
+			continue
+		}
+		for _, m := range metricFamily.Metric {
+			for _, l := range m.Label {
+				labelName := l.GetName()
+				labelNameMatched := false
+				for _, labelPattern := range acceptedLabelNames {
+					re, err := regexp.Compile(labelPattern)
+					if err != nil {
+						t.Errorf("Cannot compile pattern %v: %v", labelPattern, err)
+						continue
+					}
+					if re.MatchString(labelName) {
+						labelNameMatched = true
+						break
+					}
+				}
+				if !labelNameMatched {
+					t.Errorf("Label %v not found in documentation. Documented labels for metric %v are: %v",
+						labelName, metric, strings.Join(acceptedLabelNames, ", "))
+				}
+			}
+		}
+	}
+}
+
+// getLabelsDocumentation is a helper function that gets metric mabels documentation.
+// It returns a map where keys are metric names, and values are slices of label names,
+// and an error in case of failure.
+// By convention, UPPER_CASE parts in label names denotes wilcard patterns, used for dynamic labels.
+func getLabelsDocumentation() (map[string][]string, error) {
+	documentedMetrics := map[string][]string{}
+
+	docPath := "../../docs/"
+	docFiles, err := ioutil.ReadDir(docPath)
+	if err != nil {
+		return documentedMetrics, errors.Wrap(err, "failed to read documentation directory")
```

Comment:
I think we should return nil in the cases of error. That is the accepted pattern I believe
```suggestion
		return nil, errors.Wrap(err, "failed to read documentation directory")
```

Reply:
Sure!